SEM rewrite

JohnnyDoorn opened the pull request:
Removed the options for multiple models (discussed with Tim/Erik-Jan/Don). I will add this functionality when we have a decent (tabbed) widget for this (we will make a separate issue for this).

Commit:
First commit!

Commit:
First commit!

Commit:
most stuff works
need to add plotting and multiple models

Commit:
most stuff works
need to add plotting and multiple models

Commit:
working plots

Commit:
working plots

Commit:
Merge branch 'semRewrite' of https://github.com/JohnnyDoorn/jasp-desktop into semRewrite

Commit:
Done?

Commit:
Final comit before PR

TimKDJ commented on JASP-Engine/JASP/R/semsimple.R:
```diff
@@ -15,497 +15,201 @@
-
-  inputCorrect <- identical(errorCheck, FALSE)
-  ######################
+    semContainer <- createJaspContainer(title = "Structural Equation Modeling<br/><span style='color:#888888;font-family:monospace;font-size:12px;font-weight:normal;'>Powered by lavaan.org</span>")
```
I thought we removed this on purpose to not give lavaan a special status? (i.e., we have no "power by BayesFactor" or the like)

TimKDJ commented on JASP-Engine/JASP/R/semsimple.R:
```diff
@@ -15,497 +15,201 @@
+                            "omitResidualSingleIndicator", "residualVariances", "correlateExogenousLatents", 
+                            "addThresholds", "addScalingParameters", "correlateDependentVariables", "addPathDiagram",
+                            "outputpathdiagramstandardizedparameter", "Data", "errorCalculation", "estimator", 
+                            "emulation"))
```
This approach seems a little heavy handed. Quite a few options in here are not shared dependencies.
From what I could tell the checkboxes "Additional fit measures" until "Modification Indices/threshold" (under "Statistics") are distinct output elements that are not dependent on one another. You could put these dependencies on the respective output element itself

TimKDJ commented on JASP-Engine/JASP/R/semsimple.R:
```diff
@@ -525,487 +229,646 @@ SEMSimple <- function(dataset=NULL, options, perform="run", callback=function(..
-                           class = c("anova", "data.frame"),
-                           heading = "Chi Square Test Statistic (unscaled)\n")
+.semMainTable <- function(semContainer, options, dataset, ready) {
+  if (!ready || !is.null(semContainer[["semFitTable"]]))
```
Remove the `!ready` here and then on line 483 put (while also removing the table assignment at the end of the function) this:
```
semContainer[["semFitTable"]] <- semFitTable

if (!ready)
  return()
```
This ensures that the table has a nice empty placeholder for when the analysis is not ready (otherwise it immediately returns and all we get is whitespace in the output).

Please do this for all the tables/plot

TimKDJ replied:
This ensures we get the right hand situation after the rewrite
![placeholders](https://user-images.githubusercontent.com/18737241/68962336-9efc0880-07d4-11ea-8131-65f5f50f3773.png)

TimKDJ commented on JASP-Engine/JASP/R/semsimple.R:
```diff
@@ -525,487 +229,646 @@ SEMSimple <- function(dataset=NULL, options, perform="run", callback=function(..
+  semFitTable$addColumnInfo(name="BIC", type="number")
+  semFitTable$addColumnInfo(name="Chisq", title = "&#967;&sup2;", type="number")
+  # semFitTable$addColumnInfo(name="ChisqDiff", title = "&#916;&#967;&sup2;", type="number")
+  semFitTable$addColumnInfo(name="PrChisq", title = "p", type="number")
```
For all p values put `type = "pvalue"`, this ensures they behave properly (showing < 0.001, etc.)

TimKDJ reviewed (changes requested):
Thanks a lot for doing this Johnny. Really helped me out :)

A couple of code comments and I noticed the following while testing:
- [ ] Analysis crashes if you tick "R-squared"**
- [ ] Analysis crashes if you untick all checkboxes**
- [ ] Analysis crashes if you select a grouping variable**
- [ ] In the "Parameter Estimates" and "Modification Indices" tables the abbreviations are still base64 encoded
- [ ] The "Other Fit Indices" table no longer appears to have the "Expected Cross-Validation Index" column
- [ ] The "Covariances (lower triangle) / correlations (upper triangle)" table is (1) duplicated and (2) the table itself has no title. This only seems to be the case when you tick "Means" under "Equality Constraints"**
- [ ] If you untick "Path diagram" then the path diagram remains

** = [semRewrite.zip](https://github.com/jasp-stats/jasp-desktop/files/3852525/semRewrite.zip)

Commit:
Tim feedback

Commit:
Final update with Tim suggestions

JohnnyDoorn commented:
The error handling was not working correctly, so I think everything should work now:
-  In the "Parameter Estimates" and "Modification Indices" tables the abbreviations are still base64 encoded 
Don told me this is going to be fixed in general so Im going to leave this for now
- The "Other Fit Indices" table no longer appears to have the "Expected Cross-Validation Index" column
Lavaan does not provide this value (anymore?)

I also like the powered by lavaan. I think it should be the case that other analyses copy this style and make the R-packages more prominent, rather than the other way aorund

Commit:
Removed the "powered by Lavaan"
The plan is to add this back later, when we do this for all R-packages used in JASP

TimKDJ commented:
This is merged in https://github.com/jasp-stats/jasp-desktop/pull/3692 (I rebased your branch and made some small (encoding) fixes)
